File: app/src/main/java/com/mapsted/sample/activities/SampleMapWithAButtonActivity.java
package com.mapsted.sample.activities;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.snackbar.BaseTransientBottomBar;
import com.google.android.material.snackbar.Snackbar;
import com.mapsted.SdkError;
import com.mapsted.map.MapstedMapApi;
import com.mapsted.map.models.interfaces.OnSetSelectedPropertyListener;
import com.mapsted.map.models.layers.BaseMapStyle;
import com.mapsted.map.views.MapPanType;
import com.mapsted.map.views.MapstedMapRange;
import com.mapsted.positioning.MapstedInitCallback;
import com.mapsted.positioning.core.utils.common.Params;
import com.mapsted.sample.R;
import com.mapsted.ui.map.processing.CustomParams;
import com.mapsted.ui.map.processing.MapstedSdkController;

public class SampleMapWithAButtonActivity extends AppCompatActivity {

    private static final String TAG = SampleMapWithAButtonActivity.class.getSimpleName();
    private View rootView;
    private FrameLayout fl_map_content;
    private FrameLayout fl_map_ui_tool;


    private MapstedSdkController sdkController;

    private int dubaiMallPropertyId = 592;
    private int soukPropertyId = 600;
    private String myTag;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.i(TAG, "::onCreate");
        setContentView(R.layout.activity_sample_main);
        rootView = findViewById(R.id.rootView);
        fl_map_content = findViewById(R.id.my_map_container);
        fl_map_ui_tool = findViewById(R.id.my_map_ui_tool);
        sdkController = MapstedSdkController.getInstance();
        Params.initialize(this);
        setupMapstedSdk();
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        Log.d(TAG, "onRequestPermissionsResult: ");
        if (!MapstedSdkController.getInstance().onRequestPermissionsResult(requestCode, permissions, grantResults)) {
            super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        }
    }


    public void setupMapstedSdk() {
        Log.i(TAG, "::setupMapstedSdk");

        // TODO: This needs to be in the app itself not in app template...
        CustomParams.newBuilder()
                .setBaseMapStyle(BaseMapStyle.DARK)
                .setMapPanType(MapPanType.RESTRICT_TO_SELECTED_PROPERTY)
                .setMapZoomRange(new MapstedMapRange(6.0f, 24.0f))
                .build();

        sdkController.initializeMapstedSDK(this, fl_map_ui_tool, fl_map_content, new MapstedInitCallback() {

            @Override
            public void onSuccess() {
                Log.i(TAG, "::setupMapstedSdk ::onSuccess");
                String tag = "com.example.view.mybuttontag";
                View inflate = LayoutInflater.from(SampleMapWithAButtonActivity.this).inflate(R.layout.sample_button, null, false);
                inflate.findViewById(R.id.btn_hello).setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        String message = "Hello! You clicked " + ((Button) v).getText() + ".";
                        /*Log.d(TAG, "onClick: ");*/
                        Snackbar.make(rootView, message, BaseTransientBottomBar.LENGTH_SHORT).show();
                    }
                });
                myTag = sdkController.addViewToMap(tag, inflate);
                Log.d(TAG, "onSuccess: myTag= " + myTag);

                MapstedMapApi.selectPropertyAndDrawIfNeeded(dubaiMallPropertyId, new OnSetSelectedPropertyListener() {
                    @Override
                    public void onSetSelectedProperty(boolean isSuccessful) {
                        Log.i(TAG, "::selectPropertyAndDrawIfNeeded onSetSelectedProperty " + (isSuccessful ? "SUCCESSFUL" : "FAILED"));
                    }
                });
            }

            @Override
            public void onFailure(SdkError sdkError) {
                Log.e(TAG, "::setupMapstedSdk ::onFailure message=" + sdkError.errorMessage);
            }


        });
    }

    @Override
    protected void onStop() {
        Log.d(TAG, "onStop: ");
        sdkController.removeViewFromMap(myTag);
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        Log.d(TAG, "onDestroy: ");
        sdkController.onDestroy();
        super.onDestroy();
    }
}

File: Sample_Kotlin_Project/app/src/main/java/com/mapsted/sample_kt/activities/SampleMapWithAButtonActivity.kt
package com.mapsted.sample_kt.activities

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.google.android.material.snackbar.Snackbar
import com.mapsted.map.MapApi
import com.mapsted.map.views.MapPanType
import com.mapsted.map.views.MapstedMapRange
import com.mapsted.positioning.MapstedInitCallback
import com.mapsted.positioning.SdkError
import com.mapsted.positioning.core.utils.common.Params
import com.mapsted.sample_kt.R
import com.mapsted.sample_kt.databinding.ActivitySampleMainBinding
import com.mapsted.ui.CustomParams
import com.mapsted.ui.MapUiApi
import com.mapsted.ui.MapstedMapUiApiProvider
import com.mapsted.ui.MapstedSdkController

class SampleMapWithAButtonActivity : AppCompatActivity(), MapstedMapUiApiProvider {

    private val TAG = SampleMapWithAButtonActivity::class.java.simpleName
    private lateinit var mBinding: ActivitySampleMainBinding
    private lateinit var mapUiApi: MapUiApi
    private var myTag: String? = null
    private var mapApi: MapApi? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_sample_main)
        mapUiApi = MapstedSdkController.newInstance(applicationContext)
        mapApi = mapUiApi.getMapApi()
        Params.initialize(this)
        setupMapstedSdk()
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<String?>,
        grantResults: IntArray
    ) {
        Log.d(TAG, "onRequestPermissionsResult: ")
        if (!mapUiApi.onRequestPermissionsResult(requestCode, permissions, grantResults)) {
            super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        }
    }

    override fun provideMapstedUiApi(): MapUiApi {
        return mapUiApi
    }

    private fun setupMapstedSdk() {
        Log.i(TAG, "::setupMapstedSdk")

        CustomParams.newBuilder()
            .setMapPanType(MapPanType.RESTRICT_TO_SELECTED_PROPERTY)
            .setShowPropertyListOnMapLaunch(true)
            .setEnablePropertyListSelection(true)
            .setMapZoomRange(MapstedMapRange(6.0f, 24.0f))
            .build()

        mapUiApi.initializeMapstedSDK(
            this,
            mBinding.myMapUiTool,
            mBinding.myMapContainer,
            object : MapstedInitCallback {
                override fun onCoreInitialized() {
                    Log.i(TAG, "::setupMapstedSdk ::onCoreInitialized")
                }

                override fun onMapInitialized() {
                    Log.i(TAG, "::setupMapstedSdk ::onMapInitialized")
                }

                override fun onSuccess() {
                    Log.i(TAG, "::setupMapstedSdk ::onSuccess")
                    val tag = "com.example.view.mybuttontag"
                    val inflate = LayoutInflater.from(this@SampleMapWithAButtonActivity)
                        .inflate(R.layout.sample_button, null, false)
                    inflate.findViewById<View>(R.id.btn_hello).setOnClickListener { v ->
                        val message = "Hello! You clicked " + (v as Button).text + "."
                        Log.d(TAG, "onClick: ");
                        Snackbar.make(
                        mBinding.rootView,
                        message,
                        Snackbar.LENGTH_SHORT
                    )
                        .show()
                    }
                    myTag = mapUiApi.addViewToMap(tag, inflate)
                    Log.d(TAG, "onSuccess: myTag= $myTag")
                }

                override fun onFailure(sdkError: SdkError) {
                    Log.e(
                        TAG,
                        "::setupMapstedSdk ::onFailure message=" + sdkError.errorMessage
                    )
                }
            })
    }
}